### PR TITLE
Tests zero-sized Sidre arrays in relay tests

### PR DESCRIPTION
**Pull request by kennyweiss**
Attempt to track down issue when loading blueprint meshes with zero-sized boundary arrays in VisIt blueprint plugin.

New datasets generated by changes to Axom's `sidre_native_layout` test in: https://github.com/LLNL/axom/pull/498

### CODE REVIEW

File: src/tests/relay/data/texample_sidre_basic_ds_demo.sidre_json [hunk 1]
```diff
@@ -44,26 +44,40 @@
...
             "buffer_id": 0,
-            "schema": "{\"dtype\":\"int64\", \"number_of_elements\": 5, \"offset\": 0, \"stride\": 8, \"element_bytes\": 8, \"endianness\": \"little\"}",
+            "schema": "{\"dtype\":\"int64\",\"number_of_elements\": 5,\"offset\": 0,\"stride\": 8,\"element_bytes\": 8,\"endianness\": \"little\"}",
+            "is_applied": 1
+          },
+          "a0_i64": 
+          {
+            "state": "BUFFER",
+            "buffer_id": 1,
+            "schema": "{\"dtype\":\"int64\",\"number_of_elements\": 0,\"offset\": 0,\"stride\": 8,\"element_bytes\": 8,\"endianness\": \"little\"}",
```
Comment (kennyweiss) [inline src/tests/relay/data/texample_sidre_basic_ds_demo.sidre_json:39]: Adds an integer array "a0_i64" with 0 elements.

File: src/tests/relay/data/texample_sidre_basic_ds_demo.sidre_json [hunk 2]
```diff
@@ -44,26 +44,40 @@
...
           {
             "state": "EXTERNAL",
-            "schema": "{\"dtype\":\"int64\", \"number_of_elements\": 5, \"offset\": 0, \"stride\": 8, \"element_bytes\": 8, \"endianness\": \"little\"}"
+            "schema": "{\"dtype\":\"int64\",\"number_of_elements\": 5,\"offset\": 0,\"stride\": 8,\"element_bytes\": 8,\"endianness\": \"little\"}"
+          },
+          "b_v0": 
+          {
+            "state": "BUFFER",
+            "buffer_id": 2,
+            "schema": "{\"dtype\":\"float64\",\"number_of_elements\": 0,\"offset\": 0,\"stride\": 8,\"element_bytes\": 8,\"endianness\": \"little\"}",
```
Comment (kennyweiss) [inline src/tests/relay/data/texample_sidre_basic_ds_demo.sidre_json:49]: Adds a new "View", "b_v0" into an existing buffer. The view has zero elements.

File: src/tests/relay/data/texample_sidre_basic_ds_demo.sidre_conduit_json [hunk 3]
```diff
@@ -10,13 +10,45 @@
         {
           "i64": 
           {
-            "state": {"dtype":"char8_str", "number_of_elements": 7, "offset": 0, "stride": 1, "element_bytes": 1, "endianness": "little", "value": "SCALAR"},
-            "value": {"dtype":"int64", "number_of_elements": 1, "offset": 0, "stride": 8, "element_bytes": 8, "endianness": "little", "value": 1}
+            "state":
```
Comment (kennyweiss) [inline src/tests/relay/data/texample_sidre_basic_ds_demo.sidre_conduit_json]: I'm not sure why the formatting changed here.  Please see the `sidre_json` file for a better view of the diffs.  

These files were generated by axom's `sidre_native_layout` test, after adding a View's with zero elements.
See: https://github.com/LLNL/axom/pull/498

**Comment by kennyweiss** [comment]
This PR attempted to track down issues with zero-sized arrays.  

The new tests indicate that conduit is happy with these zero-sized arrays (i.e. it was able to load everything without segfaulting), so the issue must be elsewhere.

File: src/tests/relay/t_relay_io_handle_sidre.cpp [hunk 4]
```diff
@@ -48,6 +48,7 @@ TEST(conduit_relay_io_handle, test_sidre_basic)
     // create an equiv conduit tree for testing
     //
 
+    conduit::int64    conduit_vals_0[]  = {};
```
Comment (kennyweiss) [inline src/tests/relay/t_relay_io_handle_sidre.cpp]: Appveyor was not happy with this line:
```
  c:\projects\conduit\src\tests\relay\t_relay_io_handle_sidre.cpp(51): error C2466: cannot allocate an array of constant size 0 [C:\projects\conduit\build\tests\relay\t_relay_io_handle_sidre.vcxproj]
```
See: https://ci.appveyor.com/project/cyrush/conduit/builds/38332028/job/9u4dj5d7edmp7d6a#L3935

File: src/tests/relay/data/texample_sidre_basic_ds_demo.sidre_conduit_json (hunk 3 above)
Comment (kennyweiss) [reply src/tests/relay/data/texample_sidre_basic_ds_demo.sidre_conduit_json]: I linted the sidre json files to make it easier to read, but can back out if you don't like it

**Review by cyrush — APPROVED** [review]

**Comment by cyrush** [comment]
@kennyweiss looks good, merging -- thanks for the new test case!

**Pull request merged**